api/workload/v2/stream.go

```diff
@@ -72,13 +74,13 @@ func newStreamManager(ctx context.Context, logger *logrus.Logger, addr string, f
 		addr:           addr,
 		reconnectChan:  make(chan struct{}, 1),
 		ConnectionChan: make(chan bool, 1),
-		forceStart:     forceStart,
 	}, nil
 }
 
 // Reconect informs the stream manager that the current stream is unusable.
 func (c *streamManager) Reconnect() {
 	c.reconnectChan <- struct{}{}
+	c.ConnectionChan <- false
```

Comment:
do this first I think this potentially has ordering issues.

Reply:
+1